kytos/core/auth.py

```diff
@@ -136,232 +240,113 @@ def register_core_auth_services(self):
+        """Get JSON from request"""
+        try:
+            metadata = request.json
+            content_type = request.content_type
+        except BadRequest as err:
+            result = 'The request body is not a well-formed JSON.'
+            raise BadRequest(result) from err
+        if content_type is None:
+            result = 'The request body is empty.'
+            raise BadRequest(result)
+        if metadata is None:
+            if content_type != 'application/json':
+                result = ('The content type must be application/json '
+                          f'(received {content_type}).')
             else:
-                response = {
-                    "answer": {"users": boxes},
-                    "code": HTTPStatus.OK.value,
-                }
-
-        content = {
-            "namespace": self.namespace,
-            "callback": _list_users_callback,
-        }
-        event = KytosEvent(name="kytos.storehouse.list", content=content)
-        self.controller.buffers.app.put(event)
-        while True:
-            time.sleep(0.1)
-            if response:
-                break
-        return response["answer"], response["code"]
+                result = 'Metadata is empty.'
+            raise UnsupportedMediaType(result)
+        return metadata
 
     @authenticated
     def _create_user(self):
-        """Save a user using Storehouse."""
-        response = {}
-
-        def _create_user_callback(_event, box, error):
-            nonlocal response
-            if not box:
-                response = {
-                    "answer": 'User already exists',
-                    "code": HTTPStatus.CONFLICT.value,
-                }
-            elif error:
-                response = {
-                    "answer": "User has not been created",
-                    "code": HTTPStatus.INTERNAL_SERVER_ERROR.value,
-                }
-            else:
-                response = {
-                    "answer": "User successfully created",
-                    "code": HTTPStatus.OK.value,
-                }
-
-        req = request.json
-        password = req["password"].encode()
-        data = {
-            "username": req["username"],
-            "email": req["email"],
-            "password": hashlib.sha512(password).hexdigest(),
-        }
-        content = {
-            "namespace": self.namespace,
-            "box_id": data["username"],
-            "data": data,
-            "callback": _create_user_callback,
-        }
-        event = KytosEvent(name="kytos.storehouse.create", content=content)
-        self.controller.buffers.app.put(event)
-        while True:
-            time.sleep(0.1)
-            if response:
-                break
-        return response["answer"], response["code"]
+        """Save a user using MongoDB."""
+        try:
+            user = self.user_controller.create_user(self._get_request())
+        except (ValidationError, DuplicateKeyError) as e:
```

Comment:
Since this is a request handler, at this point the exceptions that were bubbled up need to be handled, so I'll recommend that you add individual `except` for each exception and then raise the expected `werkzeug` exception, `BadRequest` for `ValidationError` and `Conflict` for `DuplicateKeyError`. As you currently have here, I don't think the `if not user` would execute since exceptions are being re-raised.

Thanks for also updating the status code to 201

Reply:
No problem, now the exceptions are individually raised.